File: 349-intersection-of-two-arrays/intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> map=new HashSet<>();
        for(int n:nums1){
            map.add(n);
        }
        HashSet<Integer> result=new HashSet<>();
        for(int n:nums2){
            if(map.contains(n)){
                result.add(n);
            }
        }
        int[] res=new int[result.size()];
        int i=0;
        for(int n:result){
            res[i++]=n;
        }
        return res;
    }
}

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        ans=[]
        kick=set()
        for num in nums1:
            kick.add(num)
        for num in nums2:
            if num in kick and num not in ans:
                ans.append(num)
        return ans
        